examples/pytorch/application/qwen_doc_search_QA_based_on_dashscope.ipynb

```diff
@@ -282,153 +300,386 @@
+      "```\n",
+      "我的问题是：西安交大的校训是什么？。\n",
       "\n",
-      "\t```\n",
-      "\t我的问题是：清华博士发生了什么？。\n",
-      "    \n",
-      "question: 清华博士发生了什么？\n",
-      "answer: 8月23日上午9时40分，一辆由北向南行驶的金杯面包车撞到高速公路护栏上，车上5名清华大学博士后研究人员及1名司机受伤，被紧急送往沧州二医院抢救。\n"
+      "tensor([1., 2.], device='cuda:0')\n"
      ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "'西安交大的校训是“求实学、务实业”。'"
+      ]
+     },
+     "execution_count": 6,
+     "metadata": {},
+     "output_type": "execute_result"
     }
    ],
    "source": [
-    "# test the case with knowledge\n",
-    "context = search_relevant_context(question, topk=1)\n",
-    "answer = answer_question(question, context)\n",
-    "print(f'question: {question}\\n' f'answer: {answer}')"
+    "import os\n",
+    "from abc import ABC\n",
+    "from typing import Any, List, Optional, Dict, cast\n",
```

Comment:
this cell is too long to read, pls split this cell to multiple cells at each location when you need to write comment, and create a markdown block to write that comment